### ISSUE TITLE
IDISP008 should not warn when assigning this with identity in struct builder

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#477]
```cs
this = this.Append(1);
```
